File: lib/models/scene/ground.dart
import 'package:playx_3d_scene/models/scene/geometry/size.dart';
import 'package:playx_3d_scene/models/shapes/plane.dart';

class Ground extends Plane {
  double width;
  double height;
  bool isBelowModel = true;

  Ground(
      {required this.width,
      required this.height,
      super.centerPosition,
      super.normal,
      this.isBelowModel = false,
      super.material})
      : super(size: PlayxSize(x: width, z: height), id: 0);

  Map<String, dynamic> toJson() => {
        'centerPosition': centerPosition?.toJson(),
        'normal': normal?.toJson(),
        'isBelowModel': isBelowModel,
        'size': size.toJson(),
        'material': material?.toJson(),
      };
}

File: android/src/main/kotlin/io/sourcya/playx_3d_scene/core/models/scene/Ground.kt
package io.sourcya.playx_3d_scene.core.models.scene

import io.sourcya.playx_3d_scene.core.models.scene.material.Material
import io.sourcya.playx_3d_scene.core.models.shapes.Direction
import io.sourcya.playx_3d_scene.core.models.shapes.Plane
import io.sourcya.playx_3d_scene.core.models.shapes.Position
import io.sourcya.playx_3d_scene.core.models.shapes.Size

class Ground (
    centerPosition: Position? = null,
    size : Size? = null,
    val isBelowModel:Boolean = false,
    normal: Direction? = null,
    material: Material? = null
) : Plane(0,centerPosition,size,normal,material)